docs/TaskManager Developer Guide.md

```diff
@@ -0,0 +1,142 @@
+# TaskFactory API Documentation
+
+The `TaskFactory` class is responsible for creating tasks that can be executed by the `TaskManager`. It provides factory methods to create different types of tasks, such as HTTP and MQTT tasks and can be extended for other task types.
+
+## Current Methods
+
+### `createHttpTask`
+
+- **Description**: Creates an HTTP task that will perform an HTTP GET request to the specified URL. The task will call the provided callback with the HTTP response.
+- **Parameters**:
+  - `const String &url`: The URL to which the HTTP request will be made.
+  - `Task::ResponseCallback callback`: The callback function that will be called with the HTTP response.
+  - `Task::PreProcessCallback preProcess` (optional): A function that will pre-process the response before it is passed to the callback.
+- **Returns**: A `std::unique_ptr<Task>` representing the HTTP task.
+- **Example**:
+  ```cpp
+  auto task = TaskFactory::createHttpTask(
+      "https://example.com/api/data",
+      [](int httpCode, const String &response) {
+          if (httpCode > 0) {
+              Serial.println("Response: " + response);
+          } else {
+              Serial.println("HTTP request failed");
+          }
+      }
+  );
+  TaskManager::getInstance()->addTask(std::move(task));
+  ```
+
+---
+
+### `createMqttTask`
+
+- **Description**: Creates an MQTT task. This is currently a placeholder and does not contain any actual MQTT logic yet.
+- **Parameters**:
+  - `const String &topic`: The MQTT topic to subscribe to or publish to.
+  - `Task::ResponseCallback callback`: The callback function that will be called with the MQTT response.
+- **Returns**: A `std::unique_ptr<Task>` representing the MQTT task.
+- **Example**:
+  ```cpp
+  auto task = TaskFactory::createMqttTask(
+      "home/sensor/temperature",
+      [](int httpCode, const String &response) {
+          // Handle MQTT response
+      }
+  );
+  TaskManager::getInstance()->addTask(std::move(task));
+  ```
+
+---
+
+### `httpTask`
```

Comment:
Probably no need to document internal functions (in mardown) - code comments should be sufficient?

Reply:
im fine with removing the doc file.  Just seems like maybe theres a broader need for a general developer guide or a how to contribute guide.  I'll remove this file though

Reply:
The file is fine! (good even) I was just commenting about httpTask() - an internal function.